### PR TITLE
Create a store with assignable defaults backed by realm

### PR DESCRIPTION
**Pull request by furiner**
Just a basic store that lets us set defaults to them when necessary.

### CODE REVIEW

File: Kumi.Game/Database/IRealmBackedDefaultStore.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+﻿using Realms;
+
+namespace Kumi.Game.Database;
+
+public interface IRealmBackedDefaultStore<TModel>
```
Comment (Ryooshuu) [inline Kumi.Game/Database/IRealmBackedDefaultStore.cs:5]: Should probably inherit `IStoreDefaults<T>`?

File: Kumi.Game/Database/RealmBackedDefaultStore.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+            foreach (var item in DefaultValues)
+            {
+                if (existing.Any(e => Compare(e, item)))
+                {
+                    continue;
+                }
+
+                r.Add(item);
+            }
```
Comment (Ryooshuu) [inline Kumi.Game/Database/RealmBackedDefaultStore.cs:53]: Would be better to use transactions here

File: Kumi.Game/Database/RealmBackedDefaultStore.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    public void Reset()
+    {
+        realm.Write(r =>
+        {
+            foreach (var item in r.All<TModel>())
+            {
+                r.Remove(item);
+            }
```
Comment (Ryooshuu) [inline Kumi.Game/Database/RealmBackedDefaultStore.cs:83]: And here

**Review by Ryooshuu — CHANGES_REQUESTED** [review]

File: Kumi.Game/Database/RealmBackedDefaultStore.cs (hunk 2 above)
Comment (furiner) [reply Kumi.Game/Database/RealmBackedDefaultStore.cs:53]: [Write already starts a transation, no?](https://github.com/kikuyodev/kumi/blob/master/Kumi.Game/Database/RealmAccess.cs#L107-L142)

Comment (Ryooshuu) [reply Kumi.Game/Database/RealmBackedDefaultStore.cs:53]: 💀

**Review by Ryooshuu — APPROVED** [review]

**Pull request merged**